### PR TITLE
Prevent overwriting data on wrong keys in localstorage

### PR DESCRIPTION
**Pull request by garcus**
Issue reported in #51

### CODE REVIEW

**Comment by garcus** [comment]
I signed it!

File: app-localstorage/app-localstorage-document.html [hunk 1]
```diff
@@ -204,10 +204,10 @@
       },
 
       __onAppLocalStorageChanged: function(event) {
-        if (event.detail === this) {
+        if (event.detail.key !== this.key ||
+            event.detail.storage !== this.storage) {
```
Comment (cdata) [inline app-localstorage/app-localstorage-document.html]: Shouldn't we still want to check to see if `event.detail === this`?

**Comment by cdata** [comment]
Thanks for the contribution. I left a small piece of feedback, please address it!

**Comment by garcus** [comment]
Thanks @cdata 
Of course, no use syncing if "myself". I've added the statement.

**Comment by kritollm** [comment]
Was just going to report the issue with the " || this.key !== event.detail.key " fix, because the bug is still in the code. Probably not merged.

**Comment by tony19** [comment]
@cdata Any update on merging this?

**Comment by cherukumilli** [comment]
@cdata 
Can you please help merge this pull request?

**Comment by cdata** [comment]
@garcus Thanks for the PR. However, we do not accept changes without corresponding tests to describe behavior and cover regressions (per our contributing guidelines). Please add a test that demonstrates that the behavior is broken before the change and fixed after the change!

**Comment by flawyte** [comment]
Any update on merging this ? Cause without fix this component is basically broken when using several instances of it with different keys.

**Comment by cdata** [comment]
Sorry for the delay. This looks good.

**Pull request merged**